rpcserver.go

```diff
@@ -2643,14 +2643,6 @@ func (r *rpcServer) CloseChannel(in *lnrpc.CloseChannelRequest,
 		rpcsLog.Debugf("Target sat/kw for closing transaction: %v",
 			int64(feeRate))
 
-		// Before we attempt the cooperative channel closure, we'll
-		// examine the channel to ensure that it doesn't have a
-		// lingering HTLC.
-		if len(channel.ActiveHtlcs()) != 0 {
```

Comment:
Per [this comment](https://github.com/lightningnetwork/lnd/pull/8167#pullrequestreview-1780701979) it seems like the code should instead optionally check `len(ActiveHtlcs()) == 0`. That was at least my interpretation of what the `NoWait` flag should be

Reply:
Yeah this has been added so as not to break backwards compat.